packages/api/shared/Route.ts

```diff
@@ -69,26 +71,23 @@ export type RouteDefinition<Params, RequestBody, ResponseBody> = {
     req: RouteRequest<Params, RequestBody>,
     res: ResponseHelper<ResponseBody>
   ): Promise<void>;
-} & (RequestBody extends void
-  ? // eslint-disable-next-line @typescript-eslint/ban-types
-    {}
-  : {
-      /**
-       * The `zod` schema used to parse the request body.
-       * This schema should produce an object that matches the `RequestType`.
-       * If `RequestBody` is `void` then a schema is not required.
-       *
-       * Note that there are some schemas that typescript is unable to catch.
-       * The most common examples are:
-       * - Not adding optional properties to the schema.
-       * This won't return an error, but when those fields are provided, they won't be operated on.
-       * - Treating a property as required in the schema that is optional in the type.
-       * This will result in 422 errors if the field isn't provided.
-       * - Adding extra properties to the schema that aren't in the type.
-       * This could lead to unexpected 422 errors as well.
-       */
-      schema: ZodType<RequestBody>;
-    });
+
+  /**
+   * The `zod` schema used to parse the request body.
+   * This schema should produce an object that matches the `RequestType`.
+   * If `RequestBody` is `void` then a schema is not required.
+   *
+   * Note that there are some schemas that typescript is unable to catch.
+   * The most common examples are:
+   * - Not adding optional properties to the schema.
+   * This won't return an error, but when those fields are provided, they won't be operated on.
+   * - Treating a property as required in the schema that is optional in the type.
+   * This will result in 422 errors if the field isn't provided.
+   * - Adding extra properties to the schema that aren't in the type.
+   * This could lead to unexpected 422 errors as well.
+   */
+  schema?: ZodType<RequestBody>;
```

Comment:
The fancy typescript types that made this conditionally required broke with a discriminated union as a request type. I had to make it optional, so if we forget to add a schema, the body won't be available in the request.